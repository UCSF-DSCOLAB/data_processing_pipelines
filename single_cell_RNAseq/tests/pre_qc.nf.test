nextflow_pipeline {

    name "Test Workflow pipeline_pre_qc.nf"
    script "pipeline_pre_qc.nf"
    tag "qc-quantiles"


    def c4PathToTestFiles = "/krummellab/data1/integration_test_user/tutorial_lib_sep"

    // Assertion directories


    stage {
        copy c4PathToTestFiles
    }

    test("Pre-qc: freemuxlet and doubletfinder") {

        def metaDir = "$metaDir"
        def testDirAbsolute = metaDir + c4PathToTestFiles
        def freemuxDir = testDirAbsolute + "/freemuxlet_data"
        def demuxMethod = "freemuxlet"
        
        def finding_doublets = "/finding_doublets"
        def automated_processing = "/automated_processing"
        def demux_dir = "/freemuxlet"
        def test_directories = [finding_doublets, automated_processing, demux_dir]

        when {
            params {
                    load("./tests/inputs/pre_qc.json")
                    project_dir = testDirAbsolute
                    settings = {
                        skip_cellranger = true
                        merge_for_demux = true
                        merge_demux_dir = freemuxDir
                        demux_method = demuxMethod
                        default_qc_cuts_dir = testDirAbsolute
                        default_qc_cuts_file = "default_qc_cuts.csv"
                    }
            }
        }

        then {
            // Setup assertion directories
            def dm1_scg1 = testDirAbsolute + "/data/single_cell_GEX/processed/TEST-POOL-DM1-SCG1"
            def dm1_scg2 = testDirAbsolute + "/data/single_cell_GEX/processed/TEST-POOL-DM1-SCG2"
            def dm2_scg1 = testDirAbsolute + "/data/single_cell_GEX/processed/TEST-POOL-DM2-SCG1"
            def test_sample_paths = [dm1_scg1, dm1_scg2, dm2_scg1]
            def resultFilesList = []
            for (sample in test_sample_paths) {
                for (test_dir in test_directories) {
                    def dir_to_test = sample + test_dir
                    def dirFile = new File(dir_to_test)
                    def resultFiles = dirFile.listFiles().collect { it.name }
                    resultFilesList.addAll(resultFiles)
                    println "Processing: ${resultFilesList}"
                }
            }
            // Filter out the .pdf files
            def filteredPaths = resultFilesList.findAll { filePath ->
                     !filePath.endsWith('.pdf')
            }
            def sorted = filteredPaths.sort()
            assert snapshot(workflow, sorted).match()
        }
    }

    test("Pre-qc: demuxlet and doubletfinder") {

        def metaDir = "$metaDir"
        def testDirAbsolute = metaDir + c4PathToTestFiles
        def freemuxDir = testDirAbsolute + "/freemuxlet_data"
        def demuxMethod = "demuxlet"

        def finding_doublets = "/finding_doublets_dmx"
        def automated_processing = "/automated_processing_dmx"
        def demux_dir = "/demuxlet"
        def test_directories = [finding_doublets, automated_processing, demux_dir]
        
        when {
            params {
                    load("./tests/inputs/pre_qc.json")
                    project_dir = testDirAbsolute
                    settings = {
                        skip_cellranger = true
                        merge_for_demux = true
                        merge_demux_dir = freemuxDir
                        demux_method = demuxMethod
                        default_qc_cuts_dir = testDirAbsolute
                        default_qc_cuts_file = "default_qc_cuts.csv"
                    }
            }
        }

        then {
            // Setup assertion directories
            def dm1_scg1 = testDirAbsolute + "/data/single_cell_GEX/processed/TEST-POOL-DM1-SCG1"
            def dm1_scg2 = testDirAbsolute + "/data/single_cell_GEX/processed/TEST-POOL-DM1-SCG2"
            def dm2_scg1 = testDirAbsolute + "/data/single_cell_GEX/processed/TEST-POOL-DM2-SCG1"
            def test_sample_paths = [dm1_scg1, dm1_scg2, dm2_scg1]
            def resultFilesList = []
            for (sample in test_sample_paths) {
                for (test_dir in test_directories) {
                    def dir_to_test = sample + test_dir
                    def dirFile = new File(dir_to_test)
                    def resultFiles = dirFile.listFiles().collect { it.name }
                    resultFilesList.addAll(resultFiles)
                }
            }
            // Filter out the .pdf files
            def filteredPaths = resultFilesList.findAll { filePath ->
                     !filePath.endsWith('.pdf')
            }
            def sorted = filteredPaths.sort()
            assert snapshot(workflow, sorted).match()
        }
    }

}

