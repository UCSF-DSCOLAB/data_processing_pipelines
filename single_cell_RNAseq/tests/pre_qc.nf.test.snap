{
    "Should successfully demultiplex merged libaries using demuxlet and run doublet finder": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": "",
                "exitStatus": 0,
                "failed": false,
                "stdout": [
                    
                ],
                "errorMessage": "",
                "name": "workflow",
                "success": true
            },
            [
                "TEST-POOL-DM1-SCG1.clust1.samples.gz",
                "TEST-POOL-DM1-SCG1.clust1.samples.reduced.tsv",
                "TEST-POOL-DM1-SCG1_cutoffs.csv",
                "TEST-POOL-DM1-SCG1_doubletStat.rds",
                "TEST-POOL-DM1-SCG1_doublet_counts.csv",
                "TEST-POOL-DM1-SCG1_quantiles_pre.tsv",
                "TEST-POOL-DM1-SCG1_raw.rds",
                "TEST-POOL-DM1-SCG1_seurat_object_findingDoublets.rds",
                "TEST-POOL-DM1-SCG2.clust1.samples.gz",
                "TEST-POOL-DM1-SCG2.clust1.samples.reduced.tsv",
                "TEST-POOL-DM1-SCG2_cutoffs.csv",
                "TEST-POOL-DM1-SCG2_doubletStat.rds",
                "TEST-POOL-DM1-SCG2_doublet_counts.csv",
                "TEST-POOL-DM1-SCG2_quantiles_pre.tsv",
                "TEST-POOL-DM1-SCG2_raw.rds",
                "TEST-POOL-DM1-SCG2_seurat_object_findingDoublets.rds",
                "TEST-POOL-DM2-SCG1.clust1.samples.gz",
                "TEST-POOL-DM2-SCG1.clust1.samples.reduced.tsv",
                "TEST-POOL-DM2-SCG1_cutoffs.csv",
                "TEST-POOL-DM2-SCG1_doubletStat.rds",
                "TEST-POOL-DM2-SCG1_doublet_counts.csv",
                "TEST-POOL-DM2-SCG1_quantiles_pre.tsv",
                "TEST-POOL-DM2-SCG1_raw.rds",
                "TEST-POOL-DM2-SCG1_seurat_object_findingDoublets.rds"
            ]
        ],
        "timestamp": "2024-10-10T13:35:39.013467999"
    },
    "Should successfully demultiplex merged libaries using freemuxlet and run doublet finder": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": "",
                "exitStatus": 0,
                "failed": false,
                "stdout": [
                    
                ],
                "errorMessage": "",
                "name": "workflow",
                "success": true
            },
            [
                "TEST-POOL-DM1-SCG1.cel.gz",
                "TEST-POOL-DM1-SCG1.clust1.samples.gz",
                "TEST-POOL-DM1-SCG1.clust1.samples.reduced.tsv",
                "TEST-POOL-DM1-SCG1.clust1.vcf.gz",
                "TEST-POOL-DM1-SCG1.lmix",
                "TEST-POOL-DM1-SCG1.plp.gz",
                "TEST-POOL-DM1-SCG1.umi.gz",
                "TEST-POOL-DM1-SCG1.var.gz",
                "TEST-POOL-DM1-SCG1_barcodes_of_interest.filt.list",
                "TEST-POOL-DM1-SCG1_cutoffs.csv",
                "TEST-POOL-DM1-SCG1_doubletStat.rds",
                "TEST-POOL-DM1-SCG1_doublet_counts.csv",
                "TEST-POOL-DM1-SCG1_quantiles_pre.tsv",
                "TEST-POOL-DM1-SCG1_raw.rds",
                "TEST-POOL-DM1-SCG1_seurat_object_findingDoublets.rds",
                "TEST-POOL-DM1-SCG2.cel.gz",
                "TEST-POOL-DM1-SCG2.clust1.samples.gz",
                "TEST-POOL-DM1-SCG2.clust1.samples.reduced.tsv",
                "TEST-POOL-DM1-SCG2.clust1.vcf.gz",
                "TEST-POOL-DM1-SCG2.lmix",
                "TEST-POOL-DM1-SCG2.plp.gz",
                "TEST-POOL-DM1-SCG2.umi.gz",
                "TEST-POOL-DM1-SCG2.var.gz",
                "TEST-POOL-DM1-SCG2_barcodes_of_interest.filt.list",
                "TEST-POOL-DM1-SCG2_cutoffs.csv",
                "TEST-POOL-DM1-SCG2_doubletStat.rds",
                "TEST-POOL-DM1-SCG2_doublet_counts.csv",
                "TEST-POOL-DM1-SCG2_quantiles_pre.tsv",
                "TEST-POOL-DM1-SCG2_raw.rds",
                "TEST-POOL-DM1-SCG2_seurat_object_findingDoublets.rds",
                "TEST-POOL-DM2-SCG1.cel.gz",
                "TEST-POOL-DM2-SCG1.clust1.samples.gz",
                "TEST-POOL-DM2-SCG1.clust1.samples.reduced.tsv",
                "TEST-POOL-DM2-SCG1.clust1.vcf.gz",
                "TEST-POOL-DM2-SCG1.lmix",
                "TEST-POOL-DM2-SCG1.plp.gz",
                "TEST-POOL-DM2-SCG1.umi.gz",
                "TEST-POOL-DM2-SCG1.var.gz",
                "TEST-POOL-DM2-SCG1_barcodes_of_interest.filt.list",
                "TEST-POOL-DM2-SCG1_cutoffs.csv",
                "TEST-POOL-DM2-SCG1_doubletStat.rds",
                "TEST-POOL-DM2-SCG1_doublet_counts.csv",
                "TEST-POOL-DM2-SCG1_quantiles_pre.tsv",
                "TEST-POOL-DM2-SCG1_raw.rds",
                "TEST-POOL-DM2-SCG1_seurat_object_findingDoublets.rds"
            ]
        ],
        "timestamp": "2024-10-10T13:11:55.470499911"
    }
}